crates/sui-framework/sources/governance/validator_set.move

```diff
@@ -110,15 +115,65 @@ module sui::validator_set {
 
     // ==== functions to add or remove validators ====
 
-    /// Called by `sui_system`, add a new validator to `pending_validators`, which will be
+    /// Called by `sui_system` to add a new validator candidate.
+    public(friend) fun request_add_validator_candidate(self: &mut ValidatorSet, validator: Validator) {
```

Comment:
Do we have a limit on the number of candidates or only on the limit of active validators?

Reply:
We don't want or need to have a limit on the number of candidates (people who sign up to have empty staking pools) I believe. The limit is just on active validators.